source/production/F0.Minesweeper.Components/Logic/Cell/CellStatusTransition.cs

```diff
@@ -13,19 +14,31 @@ public CellStatusTransition(CellStatusType processState, MouseButtonType command
 			this.command = command;
 		}
 
-		public override int GetHashCode()
+		public static bool operator ==(CellStatusTransition left, CellStatusTransition right)
 		{
-			unchecked
+			if (left is null)
 			{
-				return 23 + 19 * processState.GetHashCode() + 19 * processState.GetHashCode();
+				return right is null;
 			}
+			return left.Equals(right);
 		}
 
+		public static bool operator !=(CellStatusTransition left, CellStatusTransition right)
+		{
+			return !(left == right);
+		}
+
+		public override int GetHashCode() => HashCode.Combine(processState, command);
+
 		public override bool Equals(object? obj)
 		{
-			var other = obj as CellStatusTransition;
+			return obj is CellStatusTransition
+				&& Equals(obj as CellStatusTransition);
```

Comment:
We are casting here twice. This can be reduced to a single cast with pattern matching:
Via `obj is CellStatusTransition transition`, which still evaluates to `boolean`, but additionally assigns the result to a new variable, so we can use it in `Equals(transition)`